request/src/main/java/com/vimeo/networking2/Configuration.kt

```diff
@@ -0,0 +1,272 @@
+
+        /**
+         * Specify a request time out. Defaults to [DEFAULT_TIMEOUT].
+         *
+         * @see Configuration.requestTimeoutSeconds
+         */
+        fun withRequestTimeout(requestTimeoutSeconds: Int) = apply {
+            this.requestTimeoutSeconds = requestTimeoutSeconds
+        }
+
+        /**
+         * Enable or disable certificate pinning. Defaults to true.
+         *
+         * @see Configuration.isCertPinningEnabled
+         */
+        fun withCertPinning(certPinningEnabled: Boolean) = apply { this.isCertPinningEnabled = certPinningEnabled }
+
+        /**
+         * Specify a log delegate, or none at all. Defaults to [DefaultLogDelegate] which logs to the system output.
+         *
+         * @see Configuration.logDelegate
+         */
+        fun withLogDelegate(logDelegate: LogDelegate?) = apply { this.logDelegate = logDelegate }
+
+        /**
+         * Specify a log level. Defaults to [LogDelegate.Level.DEBUG].
+         *
+         * @see Configuration.logLevel
+         */
+        fun withLogLevel(logLevel: LogDelegate.Level) = apply { this.logLevel = logLevel }
+
+        /**
+         * Specify a cache directory. Defaults to no cache.
+         *
+         * @see Configuration.cacheDirectory
+         */
+        fun withCacheDirectory(cacheDirectory: File) = apply { this.cacheDirectory = cacheDirectory }
+
+        /**
+         * Specify a maximum cache size. Defaults to [DEFAULT_CACHE_SIZE].
+         *
+         * @see Configuration.cacheMaxSizeBytes
+         */
+        fun withCacheMaxSizeBytes(cacheMaxSizeBytes: Long) = apply { this.cacheMaxSizeBytes = cacheMaxSizeBytes }
+
+        /**
+         * Specify a maximum cache age. Defaults to [DEFAULT_CACHE_MAX_AGE].
+         *
+         * @see Configuration.cacheMaxAgeSeconds
+         */
+        fun withCacheMaxAgeSeconds(cacheMaxAgeSeconds: Int) = apply { this.cacheMaxAgeSeconds = cacheMaxAgeSeconds }
+
+        /**
+         * Build the configuration instance using the chosen modifications and unspecified default values.
+         */
+        fun build(): Configuration = Configuration(
+            baseUrl = baseUrl,
+            clientId = clientId,
+            clientSecret = clientSecret,
+            scope = scope,
+            codeGrantRedirectUri = codeGrantRedirectUrl,
+            locales = locales,
+            accountStore = accountStore,
+            networkInterceptors = networkInterceptors,
+            interceptors = interceptors,
+            userAgent = userAgent,
+            requestTimeoutSeconds = requestTimeoutSeconds,
+            isCertPinningEnabled = isCertPinningEnabled,
+            logDelegate = logDelegate,
+            logLevel = logLevel,
+            cacheDirectory = cacheDirectory,
+            cacheMaxSizeBytes = cacheMaxSizeBytes,
+            cacheMaxAgeSeconds = cacheMaxAgeSeconds
+        )
+    }
+
+    companion object {
```

Comment:
Optional: With the  `Configuration` constructor being public I think these could also be public so if the consumer uses the constructor they still have access to the defaults without having to define them themselves?

Reply:
I think that's a good idea. They might not be necessary, but I don't want to assume too many use cases and perhaps users will need them.